osu.Game.Rulesets.Osu/Edit/Blueprints/Sliders/Components/PathControlPointVisualiser.cs

```diff
@@ -142,6 +148,23 @@ public void OnReleased(PlatformAction action)
         {
         }
 
+        /// <summary>
+        /// Handles correction of invalid path types.
+        /// </summary>
+        private void updatePathTypes()
+        {
+            foreach (PathControlPoint segmentStartPoint in slider.Path.ControlPoints.Where(p => p.Type.Value != null))
+            {
+                if (segmentStartPoint.Type.Value != PathType.PerfectCurve)
+                    continue;
+
+                ReadOnlySpan<Vector2> points = slider.Path.PointsInSegment(segmentStartPoint).Select(p => p.Position.Value).ToArray();
+                RectangleF boundingBox = PathApproximator.CircularArcBoundingBox(points);
+                if (points.Length == 3 && boundingBox.Area >= 640 * 480)
```

Comment:
An area check seems a little wrong in this context. For instance, a 20x15359 slider would still theoretically pass it. It's probably harder to abuse this than that random example, given that the bounding box is based on an arc, but I'd say checking individual dimensions would be preferable.